sky/backends/cloud_vm_ray_backend.py

```diff
@@ -863,9 +863,10 @@ def _update_blocklist_on_error(
             # Gang scheduling failure (head node is definitely up, but some
             # workers' provisioning failed).  Simply block the zones.
             assert stderr is None, stderr
-            for zone in zones:
-                self._blocked_resources.add(
-                    launchable_resources.copy(zone=zone.name))
+            if zones:
```

Comment:
Should we check `is not None` or is this `[]`? Perhaps type the args on L847-848?

Reply:
This is for checking both `is not None` and not empty. Added the type hints. Thanks!

Reply:
The complete typing is added in #1564. We may want to prioritize it a bit as that should catch the issue like this PR. : )

Reply:
nit: Shall we do `if zones is not None`? This will adhere to styleguide and empty list would work.